### ISSUE TITLE
join_information issue

### ISSUE THREAD

**Issue by magadancito** [iquasere/MOSCA#16]
Hi dear MOSCA's programmer

First of all thanks for your program it is really very useful. Unfortunately when executing the main script I had several problems that I have been able to solve manually by reading the content of the Snakefile. But I have this join_information.py script problem that I can't solve. Install MOSCA from anaconda3 using mamba, my python version is 3.7.10 installed.

P.S. sorry for my english is not my native language

Select jobs to execute...

[Fri May 14 23:50:40 2021]
rule join_information:
    input: /home/home-server/Documentos/libreria_chile/output/Annotation/uniprotinfo.tsv, /home/home-server/Documentos/libreria_chile/output/Annotation/Chile/aligned.blast, /home/home-server/Documentos/libreria_chile/output/Annotation/Chile/reCOGnizer_results.xlsx, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Infectado4.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Infectado5.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Sano8.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Sano5.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Sano3.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Sano6.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Infectado2.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Sano2.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Infectado7.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Infectado10.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Sano1.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Infectado6.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Sano4.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Sano10.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Sano7.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Infectado8.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Infectado3.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Infectado1.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Sano9.readcounts, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/Infectado9.readcounts
    output: /home/home-server/Documentos/libreria_chile/output/MOSCA_Protein_Report.xlsx, /home/home-server/Documentos/libreria_chile/output/MOSCA_Entry_Report.xlsx, /home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/expression_matrix.tsv
    jobid: 1
    threads: 0

Job counts:
	count	jobs
	1	join_information
	1
python /share/MOSCA/scripts/join_information.py -e /home/home-server/Documentos/libreria_chile/experiments.tsv -t 0 -o /home/home-server/Documentos/libreria_chile/output -if tsv -nm TMM
MissingOutputException in line 262 of /share/MOSCA/scripts/Snakefile:
Job Missing files after 5 seconds:
/home/home-server/Documentos/libreria_chile/output/MOSCA_Protein_Report.xlsx
/home/home-server/Documentos/libreria_chile/output/MOSCA_Entry_Report.xlsx
/home/home-server/Documentos/libreria_chile/output/Metatranscriptomics/expression_matrix.tsv
This might be due to filesystem latency. If that is the case, consider to increase the wait time with --latency-wait.
Job id: 0 completed successfully, but some output files are missing. 0
  File "/home/home-server/anaconda3/envs/mosca/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 584, in handle_job_success
  File "/home/home-server/anaconda3/envs/mosca/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 259, in handle_job_success
Exiting because a job execution failed. Look above for error message
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: /home/home-server/Documentos/libreria_chile/.snakemake/log/2021-05-14T235039.772067.snakemake.log

**Comment by iquasere** [comment]
Greetings! Can you please provide the experiments and config files, and the datasets causing this bug? Only a small selection if it is enough to cause this problem!

**Comment by magadancito** [comment]
Hi, dear iquasere 
I share with you the conf and experiment data. The original data where downloaded from NCBI and filter with bowtie2 but is too big to share. 

https://github.com/magadancito/mosca_bug.git

**Comment by iquasere** [comment]
Can you share the config and experiments files, and the links to get the files, then? In the repo you shared there are no files, only the error report. Also, could you try running directly the command
```
python /share/MOSCA/scripts/join_information.py -e /home/home-server/Documentos/libreria_chile/experiments.tsv -t 1 -o /home/home-server/Documentos/libreria_chile/output -if tsv -nm TMM
```
to see if the problem is restricted to snakemake?

**Comment by magadancito** [comment]
Sorry for no attach the files, I don't know what happened. I ran the script and no happens nothing. The script run without errors but not generates the outputs files.

**Comment by iquasere** [comment]
Can you attach the files now, so I can reproduce this problem in my system? Files or links of download, for the bigger files

**Comment by iquasere** [comment]
Also share the output folder, so I can skip some steps of analysis

**Comment by magadancito** [comment]
Hi of course I will send the data. I have read all the scripts called by Snakefile and I think the problem is that each script (including join_information.py) cannot write the output file at the end of the execution. For example, the preprocess.py script could not generate the final files {} _fordware.fq and {} _reverse.fq, I had to generate them manually and then the following script could already be executed without marking an error. At each step I have manually generated the outputs, but the join_information.py script is stuck not fully understanding what the script does. I hope this information helps you, greetings.
